@navikt/core/react/src/form/virtualfocus/VirtualFocus.tsx

```diff
@@ -0,0 +1,251 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  forwardRef,
+  useId,
+  useState,
+} from "react";
+import { createContext } from "../../util/create-context";
+import { useMergeRefs } from "../../util/hooks";
+import { createDescendantContext } from "../../util/hooks/descendants/useDescendant";
+import { SlottedDivElementRef } from "./SlottedDivElement";
+
+export const [
+  VirtualFocusDescendantsProvider,
+  useVirtualFocusDescendantsContext,
+  useVirtualFocusDescendantInitializer,
+  useVirtualFocusDescendant,
+] = createDescendantContext<
+  SlottedDivElementRef,
+  {
+    handlePick: () => void;
+    handleOnActive: () => void;
+  }
+>();
+
+const [VirtualFocusInternalContextProvider, useVirtualFocusInternalContext] =
+  createContext<{
+    virtualFocusIdx: number;
+    setVirtualFocusIdx: Dispatch<SetStateAction<number>>;
+    loop: boolean;
+    uniqueId: string;
+    containerRole: Props["containerRole"];
+  }>();
+
+type Props = {
+  children: React.ReactNode;
+  /**
+   * The role of the container. This is a limited subset of roles that
+   * require manual focus management.
+   *
+   * Children that are to get focus inside this container element shall be
+   * pointed to by `aria-activedescendant`.
+   **/
+  containerRole:
+    | "combobox"
+    | "grid"
+    | "listbox"
+    | "menu"
+    | "menubar"
+    | "radiogroup"
+    | "tree"
+    | "treegrid"
+    | "tablist";
```

Comment:
Since `containerRole` is only used within Anchor, i think it can be moved to the anchor-props, renamed to `role` and just set as required.